### PR TITLE
Add numeric user ID support for oauth connector

### PR DESCRIPTION
**Pull request by tsl0922**
Signed-off-by: Shuanglei Tao <tsl0922@gmail.com>

#### What this PR does / why we need it

Add numeric user ID support for oauth connector

#### Does this PR introduce a user-facing change?

```release-note
Add numeric user ID support for oauth connector
```

### CODE REVIEW

File: connector/oauth/oauth.go [hunk 1]
```diff
@@ -209,12 +209,12 @@ func (c *oauthConnector) HandleCallback(s connector.Scopes, r *http.Request) (id
...
 	}
 
-	userID, found := userInfoResult[c.userIDKey].(string)
+	userID, found := userInfoResult[c.userIDKey]
 	if !found {
 		return identity, fmt.Errorf("OAuth Connector: not found %v claim", c.userIDKey)
 	}
 
-	identity.UserID = userID
+	identity.UserID = fmt.Sprintf("%v", userID)
```
Comment (nabokihms) [inline connector/oauth/oauth.go:217]: It seems ok for id to be a number or string, yet this piece of the code will also convert arrays and maps to strings.

I suggest using a custom logic to convert claims to strings.

```go
switch userID.(type) {
case float64, int64, string:
    identity.UserID = fmt.Sprintf("%v", userID)
default:
     return identity, fmt.Errorf("OAuth Connector: %v claim should be string or number, got %T", c.userIDKey, userID)
}
```

And one more thing. Maybe we can also apply the same logic to username and preferred username claims. It is ok to have a username like, e.g., 644587, but it is an open question.

**Review by nabokihms — CHANGES_REQUESTED** [review]
Good catch! Thank you for opening this PR. I have only one thing to discuss before we can merge this.

File: connector/oauth/oauth.go (hunk 1 above)
Comment (tsl0922) [reply connector/oauth/oauth.go]: Done.

**Comment by nabokihms** [comment]
@tsl0922, the linter throws an error. Could you please check it so we can merge this PR?

**Comment by nabokihms** [comment]
@tsl0922, the linter throws an error. Could you please check it so we can merge this PR?

**Review by nabokihms — APPROVED** [review]
lgtm, thank you!

**Comment by nabokihms** [comment]
Docker build is not working, but it looks like a problem on our side.

**Pull request merged**